polyglot/piranha/src/models/source_code_unit.rs

```diff
@@ -43,16 +44,24 @@ impl SourceCodeUnit {
   }
 
   /// Writes the current contents of `code` to the file system.
-  pub fn persist(&self) {
-    if self.code.as_str().is_empty(){
-      _ = fs::remove_file(&self.path).expect("Unable to Delete file");
-    }else{
-      fs::write(&self.path, self.code.as_str()).expect("Unable to Write file");
+  /// Based on the user's specifications, this function will delete a file if empty
+  /// and replace three consecutive newline characters with two.
+  pub(crate) fn persist(&self, piranha_arguments: &PiranhaArguments) {
+    if self.code.as_str().is_empty() {
+      if piranha_arguments.delete_file_if_empty() {
+        _ = fs::remove_file(&self.path).expect("Unable to Delete file");
+      }
+    } else {
+      let content = if piranha_arguments.delete_consecutive_new_lines() {
+        let regex = Regex::new(r"\n\s*\n(\s*\n)").unwrap();
+        regex.replace_all(&self.code(), "\n${1}").to_string()
```

Comment:
a) Will this work to replace 4 empty new lines in a row with one?
b) Will it do so efficiently? Without first substituting 4 empty lines by 3, then by 2, then by 1.

Or is there a constraint in the rest of our transformation that we will only ever insert a single new empty line that might need to be deleted?

Reply:
So, I want to give no guarantee for formatting of code after the refactoring is performed. Tools like`google-java-format` or `swift-lint` or ... could be run as a followup. 
However, formats like `strings` dont have such handy formatters. So this `regex replace` particularly fixes the formatting for the `strings` file format.  Note, by default the value of this is `false` - [ref](https://github.com/uber/piranha/pull/204/files#diff-17b1537c28ddaf25ae2355afe2161ce8a3e813eaf27d06523109a7a66bfbbbf6R123)

Reply:
updated the regex.